File: src/main/java/imaxct/service/impl/BaseService.java
package imaxct.service.impl;

import imaxct.dao.*;

import javax.annotation.Resource;

/**
 * Created by imaxct on 17-4-15.
 */
public class BaseService {

    @Resource
    protected ICourseDao courseDao;

    @Resource
    protected IDepartmentDao departmentDao;

    @Resource
    protected IStudentCourseDao studentCourseDao;

    @Resource
    protected IStudentDao studentDao;

    @Resource
    protected ITeacherDao teacherDao;

    @Resource
    protected IUserDao userDao;
}

File: src/main/java/imaxct/service/impl/BaseService.kt
package imaxct.service.impl

import imaxct.dao.*

import javax.annotation.Resource

/**
 * Created by imaxct on 17-4-15.
 */
open class BaseService {

    @Resource
    protected lateinit var courseDao: ICourseDao

    @Resource
    protected lateinit var userDao: IUserDao

    @Resource
    protected lateinit var selectionDao: ISelectionDao

    @Resource
    protected lateinit var adminDao: IAdminDao
}
